<?xml version="1.0"?>
<robot name="Delta_robot">
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.09"/>
      </geometry>
      <material name="gris">
        <color rgba="0.55 0.55 0.55 1"/>
      </material>
    </visual>
  </link>

<!--BRAZOS SUPERIORES:-->
  <link name="brazo1">
    <visual>
      <origin xyz="0.125 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.25 0.03 0.03"/> <!--la = 250-->
      </geometry>
      <material name="negro">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_brazo1" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="100.0" lower="0.0" upper="1.8" velocity="0.5"/>
    <parent link="base_link"/>
    <child link="brazo1"/>
    <origin xyz="0.0620652 0 0"/> <!--A1 = 62.0652         0         0-->
  </joint>  
  <link name="brazo2">
    <visual>
      <origin xyz="0.125 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.25 0.03 0.03"/> <!--la = 250-->
      </geometry>
      <material name="negro"/>
    </visual>
  </link>
  <joint name="base_brazo2" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="100.0" lower="0.0" upper="1.8" velocity="0.5"/>
    <parent link="base_link"/>
    <child link="brazo2"/>
    <origin xyz="-0.0310326 -0.05375 0" rpy="0 0 -2.0944"/> <!--A2 = -31.0326  -53.7500         0-->
  </joint>  
  <link name="brazo3">
    <visual>
      <origin xyz="0.125 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.25 0.03 0.03"/> <!--la = 250-->
      </geometry>
      <material name="negro"/>
    </visual>
  </link>
  <joint name="base_brazo3" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="100.0" lower="0.0" upper="1.8" velocity="0.5"/>
    <parent link="base_link"/>
    <child link="brazo3"/>
    <origin xyz="-0.0310326 0.05375 0" rpy="0 0 2.0944"/> <!--A3 = -31.0326   53.7500         0-->
  </joint>  


  <!--BRAZOS INFERIORES:-->
  <link name="barra1_a"/>
  <joint name="codo1_a" type="revolute">
    <axis xyz="0 -1 0"/>
    <limit effort="100.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="brazo1"/>
    <child link="barra1_a"/>
    <origin xyz="0.25 0 0" rpy="0 0 0"/> <!--O2 = 62.0652  -107.5000   0-->
  </joint> 
  <link name="barra1_b">
    <visual>
      <origin xyz="-0.175 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.35 0.02 0.02"/> <!--lb = 350-->
      </geometry>
      <material name="negro"/>
    </visual>
  </link>
  <joint name="codo1_b" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="100.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="barra1_a"/>
    <child link="barra1_b"/>
    <origin xyz="0 0 0" rpy="0 0 0"/> <!--O2 = 62.0652  -107.5000   0-->
  </joint> 

  <link name="barra2_a"/>
  <joint name="codo2_a" type="revolute">
    <axis xyz="0 -1 0"/>
    <limit effort="100.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="brazo2"/>
    <child link="barra2_a"/>
    <origin xyz="0.25 0 0" rpy="0 0 0"/> <!--O2 = 62.0652  -107.5000   0-->
  </joint>
  <link name="barra2_b">
    <visual>
      <origin xyz="-0.175 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.35 0.02 0.02"/> <!--lb = 350-->
      </geometry>
      <material name="negro"/>
    </visual>
  </link>
  <joint name="codo2_b" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="100.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="barra2_a"/>
    <child link="barra2_b"/>
    <origin xyz="0 0 0" rpy="0 0 0"/> <!--O2 = 62.0652  -107.5000   0-->
  </joint> 

  <link name="barra3_a"/>
  <joint name="codo3_a" type="revolute">
    <axis xyz="0 -1 0"/>
    <limit effort="100.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="brazo3"/>
    <child link="barra3_a"/>
    <origin xyz="0.25 0 0" rpy="0 0 0"/> <!--O2 = 62.0652  -107.5000   0-->
  </joint>
  <link name="barra3_b">
    <visual>
      <origin xyz="-0.175 0 0" rpy="0 0 0"/>
      <geometry>
        <box size="0.35 0.02 0.02"/> <!--lb = 350-->
      </geometry>
      <material name="negro"/>
    </visual>
  </link>
  <joint name="codo3_b" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="100.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="barra3_a"/>
    <child link="barra3_b"/>
    <origin xyz="0 0 0" rpy="0 0 0"/> <!--O2 = 62.0652  -107.5000   0-->
  </joint> 


  <!--ACTUADOR:-->
  <link name="actuador">
    <visual>
      <origin xyz="0 0 0" rpy="0 -1.57075 0"/>
      <geometry>
        <cylinder length="0.01" radius="0.02"/>
      </geometry>
    </visual>
  </link> 
  <link name="aux1"/>
  <link name="aux2"/>
  <joint name="act_x" type="prismatic">
    <parent link="base_link"/>
    <child link="aux1"/>
    <limit effort="100.0" lower="-1" upper="1" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <joint name="act_y" type="prismatic">
    <parent link="aux1"/>
    <child link="aux2"/>
    <limit effort="100.0" lower="-1" upper="1" velocity="0.5"/>
    <origin rpy="0 0 -1.57075" xyz="0 0 0"/>
  </joint>
  <joint name="act_z" type="prismatic">
    <parent link="aux2"/>
    <child link="actuador"/>
    <limit effort="100.0" lower="0" upper="1" velocity="0.5"/>
    <origin rpy="0 1.57075 0" xyz="0 0 0"/>
  </joint>



</robot>